javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -61,10 +69,20 @@ public static ClassLoader getInstance(
 
     includeEmbeddedExtensionsIfFound(extensions, javaagentFile);
 
-    extensions.addAll(parseLocation(earlyConfig.getString(EXTENSIONS_CONFIG), javaagentFile));
+    String externalExtensionsLocation = earlyConfig.getString(EXTENSIONS_CONFIG);
+    if (externalExtensionsLocation != null && !externalExtensionsLocation.isEmpty()) {
+      List<URL> externalExtensions = parseLocation(externalExtensionsLocation, javaagentFile);
+      if (externalExtensions.isEmpty()) {
+        extensionClassLoadWarningMessage =
```

Comment:
when `extensionClassLoadWarningMessage ` is generated here, can you simply log it right after? 
i don't find it necessary to have another static method `logWarningIfAny` and call it in `AgentStarterImpl`.
unless it's too early because of `loggingCustomizer.init(earlyConfig)`?

Reply:
see the comment above

```
  // this class is used early, and must not use logging in most of its methods
  // in case the extension config can not be found, we save the warning
  // message and log it later, when the logging subsystem is initialized
```